controllers/hvpa_controller.go

```diff
@@ -187,8 +184,30 @@ func (r *HvpaReconciler) getSelectorFromHvpa(instance *autoscalingv1alpha1.Hvpa)
 	return selector, nil
 }
 
+func removeFromCache(namespacedName types.NamespacedName) {
+	for i, cache := range cachedNames[namespacedName.Namespace] {
+		if cache.Name == namespacedName.Name {
+			len := len(cachedNames[namespacedName.Namespace])
+			cachedNames[namespacedName.Namespace][i] = cachedNames[namespacedName.Namespace][len-1]
+			cachedNames[namespacedName.Namespace][len-1] = nil
+			cachedNames[namespacedName.Namespace] = cachedNames[namespacedName.Namespace][:len-1]
+			log.V(3).Info("HVPA", namespacedName.Name, "removed from cache")
+			break
+		}
+	}
+}
+
 // manageCache manages the global map of HVPAs
-func (r *HvpaReconciler) manageCache(instance *autoscalingv1alpha1.Hvpa) {
+func (r *HvpaReconciler) manageCache(instance *autoscalingv1alpha1.Hvpa, namespacedName types.NamespacedName, foundHvpa bool) {
+	cacheMux.Lock()
+	defer cacheMux.Unlock()
+
+	if !foundHvpa {
+		// HVPA doesn't exist anymore, remove it from the cache
+		removeFromCache(namespacedName)
```

Comment:
Why make `manageCache` the only internal API? Why not call `removeFromCache` directly from `Reconcile` provided synchronisation is handled?

Reply:
I thought that it would be cleaner if all potential writers to the cache could only deal with 1 API instead of multiple APIs to "manage cache".
WDYT?

Reply:
Sure. Then can you please make it clear either in the comments or by explicitly making `ManageCache` public that `removeFromCache` is internal and not to be called directly.

Reply:
Sure. Will do both. Thanks.